### ISSUE TITLE
Exceptions thrown in callback are being caught and ignored

### ISSUE THREAD

**Issue by regular** [tschaub/gh-pages#20]
``` javascript
  publish(foo, bar, function(err) {
    undefined();
  });
```
will fail silently, making it hard to find errors.

**Comment by regular** [comment]
I am using this ugly workaround

``` javascript
var ghpages = require('gh-pages');

function ghpagesPublish(bundleDir, options, cb) {
    ghpages.publish(bundleDir, options, function(err) {
        try {
            cb(err);
        } catch(e) {
            console.error(e.stack);
            process.exit(1);
        }
    });
}
```

**Comment by tschaub** [comment]
Reopening since this was not actually fixed by #35.
